### PR TITLE
[WIP] M2sys for windows -- it seems that scipy does this upstream

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
The standard `libblas` build failed with
```
D:\bld\scipy_1602676883547\work/scipy/integrate/quadpack/dqc25f.f:293: undefined reference to `dgtsv_'
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,7 +22,8 @@ requirements:
     # - m2w64-gcc-fortran         # [win]
     - m2w64-toolchain            # [win]
   host:
-    - libblas
+    - openblas  # [win]
+    - libblas   # [not win]
     - libcblas
     - liblapack
```
Comment (beckermr) [inline recipe/meta.yaml:28]:
```suggestion
    - libcblas  # [not win]
    - liblapack  # [not win]
```

Comment (hmaarrfk) [reply recipe/meta.yaml]: hmm, ok, lets try

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
It is still picking up blas with MKL
```
## Package Plan ##

  environment location: D:\bld\scipy_1602680018755\_h_env


The following NEW packages will be INSTALLED:

    ca-certificates: 2020.6.20-hecda079_0       conda-forge
    certifi:         2020.6.20-py38h9bdc248_2   conda-forge
    cython:          0.29.21-py38h4b9bc1a_1     conda-forge
    intel-openmp:    2020.1-216                 defaults   
    libblas:         3.8.0-16_mkl               conda-forge
    libcblas:        3.8.0-16_mkl               conda-forge
    libflang:        5.0.0-h6538335_20180525    conda-forge
    liblapack:       3.8.0-16_mkl               conda-forge
    libopenblas:     0.3.10-pthreads_h1662909_5 conda-forge
    llvm-meta:       5.0.0-0                    conda-forge
    mkl:             2020.1-216                 defaults   
    numpy:           1.16.5-py38hc71023c_0      conda-forge
    openblas:        0.3.10-pthreads_h6ebc607_5 conda-forge
    openmp:          5.0.0-vc14_1               conda-forge
    openssl:         1.1.1h-he774522_0          conda-forge
    pip:             20.2.3-py_0                conda-forge
    pybind11:        2.5.0-py38h95a2b95_1       conda-forge
    python:          3.8.6-h60c2a47_0_cpython   conda-forge
    python_abi:      3.8-1_cp38                 conda-forge
    setuptools:      49.6.0-py38h9bdc248_2      conda-forge
    sqlite:          3.33.0-he774522_1          conda-forge
    vc:              14.1-h869be7e_1            conda-forge
    vs2015_runtime:  14.16.27012-h30e32a0_2     conda-forge
    wheel:           0.35.1-pyh9f0ad1d_0        conda-forge
    wincertstore:    0.2-py38h32f6830_1005      conda-forge

```

**Comment by hmaarrfk** [comment]
The only MKL is being used to build the numpy packages. I wonder why. There is no mention of MKL in the numpy recipe.

File: recipe/meta.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/meta.yaml]: Why?

Comment (beckermr) [reply recipe/meta.yaml]: just hacking on it - might be wrong ofc

**Comment by hmaarrfk** [comment]
I think the numpy builds need to first be completed without MKL.

**Comment by jakirkham** [comment]
We actually don't build `numpy` with `mkl`. We build it with NETLIB's `blas` and then rely on the standard BLAS API to swap out other BLAS libraries on-demand. That said, it looks like the changes to `scipy` here are not following that pattern. Possibly part of the issue that's occurring here?

**Comment by hmaarrfk** [comment]
@jakirkham sorry about all the different tries. but you can look at the results of the 1st build. They include the "standard" blas and lapack linking. They failed with a missing symbol during fortran linking.

It also seems that numpy changed how they look for linkers and compilers, so that seemed to have been the cause of failure when I unpinned numpy accidentally.

I understand that `mkl` is being pulled it since it has higher priority. I won't focus my efforts there anymore. Thank both of you and @isuruf for explaining this to me.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by hmaarrfk** [comment]
Could not find library quadpack.

Could not find library mach

**Comment by hmaarrfk** [comment]
Thanks @isuruf

**Pull request closed without merging**